### ISSUE TITLE
Provide a way to suppress tex2jax override warning

### ISSUE THREAD

**Issue by memeplex** [executablebooks/MyST-Parser#384]
**Is your feature request related to a problem? Please describe.**

```
WARNING: `mathjax_config['tex2jax']` is set, but `myst_update_mathjax = True`, and so this will be overridden. Set `myst_update_mathjax = False` if you wish to use your own config
```

I understand the rationale for the override and it's ok that I get a warning about it, but there should be a way to suppress it once it becomes clear that it's a feature I've deliberately chosen to keep.

**Describe the solution you'd like**

Any way to disable the spurious warning that you consider appropriate. TBH I prefer the warning to never be shown over the current situation.

**Describe alternatives you've considered**

`grep -v` (or similar) is not a good solution because I often use make or sphinx-autobuild, not vanilla sphinx-build.

Be constantly warned, but false positives are no good, I like to keep my build logs tidy.

**Comment by memeplex** [comment]
I also have a question regarding this, does the override apply globally or just for pages generated from markdown?

I see in the code:

```python
    if (
        app.config.mathjax_config is None
        and app.env.myst_config.update_mathjax  # type: ignore[attr-defined]
    ):
```

so my first though was to explicitly set mathjax_config and toggle the myst update off. But then I would be changing that globally, wouldn't I? So the question above, would it be any different from what myst is doing (besides suppressing the warning)?

**Comment by memeplex** [comment]
Now I believe the problem could be the `app.config.mathjax_config is None` check.

According to https://www.sphinx-doc.org/en/master/usage/extensions/math.html#confval-mathjax_config:

> The default is empty (not configured).

So empty, not None, indeed some extensions like `nbsphinx` fail to load if it's None.

I believe the check should be `not app.config.mathjax_config`, so when no configuration was provided the warning is not shown since myst is overriding nothing.

**Comment by chrisjsewell** [comment]
Heya, why not just set `myst_update_mathjax = False`?

and yes it does currently apply globally

**Comment by memeplex** [comment]
Indeed it's what I'm doing but I assume the override is somehow preferable because, well, it's being done by default. Perhaps it's more efficient or strict given that it directs mathjax not to parse unnecessary syntax that, moreover, shouldn't be interpreted as latex in the rare event that it's found? In that case I could set the override option to False and then configure mathjax_config exactly as the override would have done but then what would be the override setting for?

But AFAICS these issues vanish if you change the check in https://github.com/executablebooks/MyST-Parser/blob/fb8c8fd2d306c135841a73d4efbd078953f8f21f/myst_parser/mathjax.py#L35 as I suggested above. I would argue that the check is wrong because sphinx docs clearly state that it's an empty value, and not None, that means "unconfigured". And some extensions are unable to cope with `mathjax_config = None`, hence suggesting it's not an expected value in practice. Besides `not mathjax_config` will check also for None, just in case.

So the logic would be: if unconfigured, proceed without warning, which seems pretty sane to me. In fact, it seems it's the intention of that check.

**Comment by chrisjsewell** [comment]
Yeh no worries, thanks the feedback. I'd not that I'm already addressing this in #378 anyhow

**Comment by memeplex** [comment]
Great! Notice also that the code in that section is convoluted and repetitive.

```
    if (
        app.config.mathjax_config is None
        and app.env.myst_config.update_mathjax  # type: ignore[attr-defined]
    ):
        app.config.mathjax_config = {  ...  }  <----  config dict
    elif app.env.myst_config.update_mathjax:  # type: ignore[attr-defined]
        if "tex2jax" in app.config.mathjax_config:
            logger.warning(...)
            app.config.mathjax_config = {  ...  }  <----  the same config dict
        }
```

I might be missing something but it probably should be written more like:

```
if app.env.myst_config.update_mathjax:
   if (app.config.mathjax_config or {}).get("tex2jax"):
       logger.warning(...)
   app.config.mathjax_config = {  ...  }
```

**Comment by chrisjsewell** [comment]
I think this is now closed by 340e7197179c3720f7cdb9bef87d47c71bd204d2 (plus 05d9972f80c73bf31ef746c7eefe6c54717b26b6), but obviously let me know if you think any other improvements should be made 😄
